pkg/status/render.go

```diff
@@ -69,6 +70,9 @@ func FormatStatus(data []byte) (string, error) {
 	if traps.IsEnabled() {
 		renderStatusTemplate(b, "/snmp-traps.tmpl", snmpTrapsStats)
 	}
+	if config.IsContainerized() {
+		renderStatusTemplate(b, "/autodiscovery.tmpl", autodiscoveryErrors)
```

Comment:
Should we only render this if `len(autodiscoveryErrors) > 0`, or alternatively make the template here: `pkg/status/templates/autodiscovery.tmpl` not print the `errors` section if there are none? I feel like this could look a little weird as-is when there are no errors?

Reply:
I used empty maps for when there are no errors based on [this comment](https://github.com/DataDog/datadog-agent/pull/7906#discussion_r625785350). When there are no errors, I don't think the `Autodiscovery` section should populate in the agent status. This is what I see when I build a custom agent with my changes and there are no pod annotation errors: https://a.cl.ly/jkuP17Zb